core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -134,13 +137,12 @@ void begin(IsolationLevel level)
 	void rollback()
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result;
-		synchronized (futures) {
-			result = executor.submit(() -> {
-				txnConnection.rollback();
-				return true;
-			});
+		Future<Boolean> result = executor.submit(() -> {
+			txnConnection.rollback();
+			return true;
+		});
 
+		synchronized (futures) {
```

Comment:
I will work through this with either semaphores or putting the Future.get inside of the synchronized block. It was already possible that hasActiveOperations was being called concurrently, as the executor doesn't retain the monitor from this thread.